vignettes/articles/startup_messages.Rmd

```diff
@@ -0,0 +1,149 @@
+", file = file.path(tempdir, "startuppackage", "R", "conflict.R"))
+
+
+# Add fake dependency to dplyr
+desc <- readLines(file.path(tempdir, "startuppackage/DESCRIPTION"))
+desc <- c(desc,
+          "Depends: dplyr")
+cat(desc, sep = "\n", file = file.path(tempdir, "startuppackage/DESCRIPTION"))
+
+# Open folder for verification
+# browseURL(file.path(tempdir, "startuppackage"))
+
+# Create documentation
+roxygen2::roxygenize(file.path(tempdir, "startuppackage"))
+# Install locally
+remotes::install_local(file.path(tempdir, "startuppackage"),
+                       force = TRUE, upgrade = "never")
+
+```
+
+Now, we can load the newly created package.     
+
+```{r}
+# Detach packages
+detach("package:startuppackage", unload=TRUE)
+detach("package:dplyr", unload=TRUE)
+
+# Load library
+library(startuppackage)
+```
+
+Upon loading, {startuppackage} produces several warnings and messages that we would like to silence:   
+- a message informing that {dplyr}, a dependency of the {startuppackage}, is required      
+- a warning about the conflicts in {dplyr}     
+- a startup message specific to {startuppackage}      
+- a warning about the conflicts in {startuppackage}       
+
+
+## Loading a library without warnings
+
+To suppress the warnings about conflicts, you can set the `warn.conflict` argument to `FALSE` in `library()`. As written in the documentation of the `library()` function, a conflict is a function masking a function, or a non-function masking a non-function.
+
+```{r}
+detach("package:startuppackage", unload=TRUE)
+detach("package:dplyr", unload=TRUE)
+library(startuppackage, warn.conflicts = FALSE)
+```
+  
+Note that this removes only the warnings regarding the conflicts of the {startuppackage}, not the conflicts relative to the {dplyr} package, which is a dependency from the {startuppackage}.
+
+
+## Loading a library without messages
+
+To suppress the messages regarding the attachement of a package (either confirming the attachement of the package or warning about failed attachement), set the `quietly` argument in the `library()` function to `TRUE`. This will suppress messages relative to dependencies, but not warnings about conflicts.
+
+```{r}
+detach("package:startuppackage", unload=TRUE)
+detach("package:dplyr", unload=TRUE)
+library(startuppackage, quietly = TRUE)
+```
+
+Note that here, it only removed the message regarding the existence of the {dplyr} dependency.             
+
+
+## Loading a library silently with the function suppressPackageStartupMessages()
+
+`suppressPackageStartupMessages()` ignores diagnostic messages when loading a package. It also ignores warning messages about conflicts.
+```{r}
+detach("package:startuppackage", unload=TRUE)
+detach("package:dplyr", unload=TRUE)
+suppressPackageStartupMessages(library(startuppackage))
+```
+
+Here the `suppressPackageStartupMessages()` function suppressed all messages and warnings. 
+
+
+## Loading a library silently within a chunk
```

Comment:
You need to reveal the code a user will actually use for this if they are making a reprex:

```
#+ message = FALSE, warning = FALSE
library(whatever)

#+ reprex-body
# reprex code goes here ...
```

Reply:
That is: these chunk options *can* be used in an R script, you don't need to be writing R Markdown. And above I show the syntax.